### ISSUE TITLE
PES2017 has problems after the mipmap setting was removed. Possibly also other games.

### ISSUE THREAD

**Issue by hanooty** [hrydgard/ppsspp#9733]
In v 1.4.1 and v1.4.2
I can show black screen in any textures point when playing games 
Settings:
Buffering rendering 
How can i fix it???
Thanks

**Comment by LunaMoo** [comment]
Come on, some people here might be very talented, but likely not a single person is a fortune teller or some other mind reader.:]

 Include at least some basic info:
- about your device,
- games you're having problems with,
- and what exactly is the problem as ~ "can show black screen in any textures point" doesn't have much sense in english, it's black screen or not? If not it will be easier to just see a screenshot.;]



 Removing the setting would not affected anything unless you kept mipmaps disabled due to some actual bug likely specific to your device, that setting was also removed after 1.4.1, so if it was already broken then, it's definitely not that.
 You should find which was the first version your problems started ~ it shouldn't take much time simply by [checking builds](http://buildbot.orphis.net/ppsspp/index.php?m=fulllist) half way between the one you last remember working and one which is now broken.

**Comment by hanooty** [comment]
Hello. ..
Thanks for your reply. ..
About my device :
Android Samsung Galaxy NOTE_3 (n9005)
RAM: 3GB
CPU Information:
Armv7 processor rev 0 (v7I)
Cores: 4
GPU Informations:
3D API     OpenGL ES
Driver adreno tm 330
ÀPI VERSION : V3.0.0 ES
Resolutions :
1920×1080
●game having a problemes:
Pes 17
Waiting for screen shot 
《《》》

**Comment by hanooty** [comment]
When i play the game pes2017 the black screen just in some textures like
Crowd..nets
I'm trying to give you tutorial because i can't upload screen shot files...
I didn't found upload buttons

**Comment by unknownbrackets** [comment]
If you use an older build, and you enable mipmaps, this happens there too?  But disabling mipmaps works around it?

-[Unknown]

**Comment by unknownbrackets** [comment]
Have you tried the latest git build?

To confirm - the same scene which shows black textures on a phone does not show black on say a desktop, right?  I just want to understand if it's a "some games are showing black textures on all devices always" bug, or a "some devices are showing a certain type of texture as black" bug.  Since I do not have a warehouse of devices, I'd prefer it was the former, but it is what it is.

I don't have PES2017, either, so the more games you can find which have this problem, the easier it will be to fix.

And, to be clear, "in older versions that have a mipmap option"... as I asked above, does it work whether that setting is on or off, or does it not work?

It's possible this is some kind of rounding error, or case where we're over-specifying the available mip levels.  Or maybe it was that PowerVR bug workaround (which is why I ask about the latest git build.)

-[Unknown]

**Comment by hrydgard** [comment]
GLES is often much more of a stickler about texture completeness - very common to see black textures on ES while it's still working on desktop...   It can be as simple as doing glTexImage and specifying more mip levels than you then actually submit with glTexSubimage or enabling wrapping on a non-pow2-sized texture on ES 2.0 where that is not allowed.

**Comment by hanooty** [comment]
Ok, i'll give you full answers about your questions. ..
Yes if i used older version and i enabled mipmap option also black screen happens, but when is un-enable  (unchecked) i can play without blank screen ok...
But in new versions that's began from v 1.4.2 and up i can show black screen in alot of textures and no mipmap option to desable it...
I'm always use update builds and try another settings but loss..

**Comment by unknownbrackets** [comment]
Did this always happen with the mipmap setting on?  It sounds like the recent mipmapping changes didn't introduce this.  It's just that you can no longer turn it off.

Your device has a Adreno 330.  This is the same GPU as my Nexus 5, even.

If you have a desktop using Windows, it'd help a lot if you could follow these steps on it:
https://github.com/hrydgard/ppsspp/wiki/How-to-find-a-graphic-issue-with-the-GE-debugger

Alternatively, you could find the same area where you see the black texture on your phone, and then click "Record" (this options shows only in very recent git builds.)  Then hit "Resume", and when the game is playing again, quit.  In your "memstick/PSP/SYSTEM/DUMP/" directory will appear a file, which you can upload to Google Drive or where ever, and post a link here too.  Please also give a screenshot of your phone showing the black thing, so I can tell what thing it is causing the problem.

Basically, I'm looking for more information on what the game is trying to draw that appears black on your phone.  Even if it shows correctly on your desktop, having this information will help me try to deduce what is making the phone not render correctly, or even try it on my phone.

-[Unknown]

**Comment by hrydgard** [comment]
Is this still problematic?

**Comment by hrydgard** [comment]
PES2017 by the way? Is that a hack of PES2012?

**Comment by hrydgard** [comment]
@hanooty  is this still an issue in the latest builds?

**Comment by unknownbrackets** [comment]
This issue is being closed because it hasn't been updated with feedback.

It's hard to tell when fixes in PPSSPP might fix other games, and sometimes certain settings or cheats may cause bugs that can't be fixed.

If you have more information or can confirm it still happens in the latest git builds, please reply to this issue and it'll be reopened.  If you have a new issue with the same game, just create a new issue instead.

-[Unknown]
